torchx/c_src/torchx.cpp

```diff
@@ -3,6 +3,8 @@
 
 #include "nx_nif_utils.hpp"
 
+#define EMPTY_TAG 0xABADFACE
```

Comment:
Could it be the null byte instead?

Reply:
No. this is memory content not pointer itself.

Reply:
Does it mean we can have a false positive? Could the memory content ever be `0xABADFACE`? I would add a comment then explaining it is memory content and why this is a "safe" value to use. :)

Reply:
yes, it is possible in theory. But in reality, torch structure is a C++ object which has vtable as first element. the value here is not a valid vtable ptr.